pandas/core/base.py

```diff
@@ -1201,6 +1201,11 @@ def _map_values(self, mapper, na_action=None):
             else:
                 values = self.values
 
+            if is_categorical_dtype(values):
```

Comment:
put this before the is_extension_type test as a categorical IS an extension type.